.github/workflows/ci.yml

```diff
@@ -0,0 +1,57 @@
+name: ci
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+env:
+  PUPPETEER_SKIP_DOWNLOAD: true
+
+jobs:
+  build-lib:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ ubuntu-latest ]
+        node: [ 16 ]
+
+    defaults:
+      run:
+        working-directory: source/packages/ui-library
+
+    steps:
+      - name: Checkout 🛎
+        uses: actions/checkout@v3
+
+      - name: Setup node env 🏗
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node }}
+          cache: 'npm'
+          cache-dependency-path: package-lock.json
+
+      - name: Get npm cache directory
+        id: npm-cache-dir
+        shell: bash
+        run: echo "dir=$(npm config get cache)" >> ${GITHUB_OUTPUT}
+
+      - name: Cache node_modules 📦
+        uses: actions/cache@v3
+        id: npm-cache # use this to check for `cache-hit` ==> if: steps.npm-cache.outputs.cache-hit != 'true'
+        with:
+          path: ${{ steps.npm-cache-dir.outputs.dir }}
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+
+      - name: Run installation 👨🏻‍💻
+        run: npm ci
```

Comment:
Wenn wir hier npm ci gibt es einen clean install brauchen wir dann überhaupt den cache im step vorher oder wie funktioniert das genau?

Reply:
ich meine mehr ob `npm ci` diesen Cache invalidiert bzw. löscht 
```
If a node_modules is already present, it will be automatically removed before npm ci begins its install.
```
[npm ci](https://docs.npmjs.com/cli/v7/commands/npm-ci)

Reply:
Nein das sollte nicht stattfinden, da das Cache-Directory und das `node_modules` Directory nicht das gleiche sind. Er würde das Directory `node_modules` löschen und dann mit den Packages aus dem Cache und falls nicht vorhanden von remote jeweils wieder erstellen. Der Cache sollte die Anzahl Remote-Aufrufe aufs Registry reduzieren. 

Auf jeden Fall ist nichts in der npm-Dokumentation zu finden, was darauf hinweist das `npm ci` neben den `node_modules` auch den Cache invalidiert oder cleant. 
https://docs.npmjs.com/cli/v9/commands/npm-ci#description